### ISSUE TITLE
scPRAM model training on Kang data

### ISSUE THREAD

**Issue by MinaKarimpour** [jiang-q19/scPRAM#2]
Recently, I was running the scPRAM model on Kang data obtained from (Lotfollahi et al. 2023) paper. I followed the scPRAM tutorial pipeline to generate predictions for the cell type in the OOD setting. However, I have encountered a discrepancy with the loss values during training, which appear to be higher than those reported in the tutorial.

For example, my training loss values are as follows:
Epoch 1: SCPRAM_loss = 1.77e+3, kl_loss = 323, recon_loss = 3.54e+3
Epoch 5: SCPRAM_loss = 1.66e+3, kl_loss = 612, recon_loss = 3321.25
Epoch 11: SCPRAM_loss = 1.41e+3, kl_loss = 1.53e+3, recon_loss = 2.82e+3
Epoch 25: SCPRAM_loss = 1.48e+3, kl_loss = 2.61e+3, recon_loss = 2.95e+3
Epoch 51: SCPRAM_loss = 1.50e+3, kl_loss = 2.55e+3, recon_loss = 2.99e+3
Epoch 75: SCPRAM_loss = 1.50e+3, kl_loss = 2.19e+3, recon_loss = 2.99e+3
Epoch 99: SCPRAM_loss = 1.35e+3, kl_loss = 1.83e+3, recon_loss = 2.69e+3

Given that the losses remain consistently high throughout the training process, could there be an underlying issue? 

PS. I used the scpram.data_process function to preprocess the count data before passing it to the scPRAM model.

**Comment by jiang-q19** [comment]
Hello, I noticed that during the training process, the reconstruction loss is decreasing, but the KL divergence is increasing significantly, which leads to an increase in the overall loss function. You can consider reducing the weight of the KL divergence (self.kl_weight). If you are concerned about the increase in the loss function, you can also consider adjusting the learning rate during the training process. Additionally, please ensure that you are applying the processing function to the original count data, rather than reprocessing data that has already been processed. Lastly, I have provided a pre-processed PBMC dataset for direct use at this link:https://github.com/jiang-q19/scPRAM/blob/main/train_pbmc.h5ad
